src/snowflake/snowpark/_internal/server_connection.py

```diff
@@ -244,9 +244,18 @@ def _get_string_datum(self, query: str) -> Optional[str]:
     @SnowflakePlan.Decorator.wrap_exception
     def get_result_attributes(self, query: str) -> List[Attribute]:
         return convert_result_meta_to_attribute(
-            run_new_describe(self._cursor, query), self.max_string_size
+            self._run_new_describe(self._cursor, query), self.max_string_size
         )
 
+    def _run_new_describe(self, cursor: SnowflakeCursor, query: str) -> Union[List[ResultMetadata], List["ResultMetadataV2"]]:
+        result_metadata = run_new_describe(cursor, query)
+
+        describe_listeners = [listener for listener in self._query_listener if hasattr(listener, 'describe_listener')]
+        for listener in describe_listeners:
+            listener._add_query(QueryRecord(cursor.sfqid, query, True))
```

Comment:
This has an impact to existing customers. How about add an optional parameter `session.query_history(include_describe: bool=False)`?

Reply:
Updated the QueryHistory with include_describe